internal/sysfs/sock_go.go

```diff
@@ -0,0 +1,56 @@
+//go:build !tinygo
```

Comment:
will sockets be used at all in the tinygo version? If they won't maybe it's worth keeping `sock.go` the default (`//go:build !tinygo`) and then stub everything in `sock_tinygo.go` (`//go:build tinygo`)

I am afraid all these bifurcated code paths might become confusing

Notice that there is already a `sock_unsupported.go` with heading:

```
//go:build !linux && !darwin && !windows
```

maybe it's enough to add `|| tinygo` there?

Reply:
> will sockets be used at all in the tinygo version?

I hope so, yes.

Reply:
> If they won't maybe it's worth keeping sock.go the default (//go:build !tinygo) and then stub everything in sock_tinygo.go (//go:build tinygo)

That sounds like a good idea, since I do not know timeframe for the sockets support.